link/internal/services/machine.go

```diff
@@ -0,0 +1,95 @@
+package services
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	logger "linkSM/internal/utils"
+	"log/slog"
+)
+
+type createMachineParams struct {
+	Name             string `json:"name" `
+	Isrepairrequired bool   `json:"Isrepairrequired",omitempty`
+	Isactive         bool   `json:"Isactive",omitempty`
+}
+
+func CreateMachine(data []byte, log *slog.Logger, url string) ([]byte, error) {
+	log.Info("Start processing shift creation request")
+
+	// map[string]interface{}, because the request may not match the structure
+	var rawData map[string]interface{}
+	if err := json.Unmarshal(data, &rawData); err != nil {
+		log.Error("Failed to unmarshal input data", logger.ErrToAttr(err))
+		return nil, fmt.Errorf("input data parsing failed: %w", err)
+	}
+
+	// validation required fields
+	name, err := getString(rawData, "name")
+	if err != nil {
+		return nil, fmt.Errorf("validation error: %w", err)
+	}
+
+	machine := createMachineParams{
+		Name: name,
+	}
+	//validation optional fields
+	if val, ok := rawData["isrepairrequired"]; ok {
+		if isRepair, err := getBool(val); err == nil {
+			machine.Isrepairrequired = isRepair
+		} else {
+			log.Warn("Invalid isrepairrequired value", logger.ErrToAttr(err))
+		}
+	}
```

Comment:
as I remember GinGonic and other frameworks can simplify all this validations logic - just create structure with pointers in optional fields and after unmarshalling you will have nil there. It will be much more simplier approach!